### ISSUE TITLE
C BINDING : Segfault when using the set_url fonction

### ISSUE THREAD

**Issue by Tch3renk0v** [saucer/saucer#23]
### ✍️ Bug Description

The obtained exe segfault at exit (closing the window).

### 🔁 Steps to reproduce

Minimal reproducible code : 
"""
`#include <stdio.h>

#include <saucer/webview.h>
#include <saucer/window.h>

int main()
{

    struct saucer_options* options = saucer_options_new();
    struct saucer_handle* h = saucer_new(options);

    char url[] = "https://google.com";
    saucer_webview_set_url(h, url);
    saucer_window_show(h);
    saucer_window_run(h);

    saucer_free(h);
    saucer_options_free(options);
    
    return 0;
}`
"""

### ❔ Expected behavior

No segfault.

### 🛸 Saucer Version

C binding : 60985ffdc41c6c8d5398957ec9e8fa1ee3881f08

### 🖥️ Operating System

Linux

### 🖥️ Distribution

Archlinux, Qt5 backend

### ✅ I confirm that...

- [X] I have collected all the required information and read all the comments in this document
- [X] I searched for an existing bug report for this issue
- [X] the problem does occur with the reproduction steps I provided
- [X] what I described is indeed a problem

**Comment by Curve** [comment]
The c-bindings are not yet full complete but `set_url` works fine in Java bindings o.O

**Comment by Tch3renk0v** [comment]
Thanks for this reply !
I will then wait for the latest fixes to be merged before retrying.
A quick analysis show that it's indeed coming from Qt5's libQt5WebEngineCore.so.5.

**Comment by Curve** [comment]
> Thanks for this reply !
> I will then wait for the latest fixes to be merged before retrying.
> A quick analysis show that it's indeed coming from Qt5's libQt5WebEngineCore.so.5.

Good idea :) I'll let you know once the latest changes are mainlined (will come with a new release) - We can then confirm if the issue is fixed ^^

**Comment by Tch3renk0v** [comment]
With pleasure, just let me know.

If it helps, I also tested serving embedded files with no issue.
